src/shared/api.ts

```diff
@@ -63,7 +63,17 @@ export const shouldUseReasoningEffort = ({
 }: {
 	model: ModelInfo
 	settings?: ProviderSettings
-}): boolean => (!!model.supportsReasoningEffort && !!settings?.reasoningEffort) || !!model.reasoningEffort
+}): boolean => {
+	// If enableReasoningEffort is explicitly set to false, reasoning should be disabled
+	if (settings?.enableReasoningEffort === false) {
```

Comment:
Roomote makes a good point about the default. Should this be:
```suggestion
	if (!settings?.enableReasoningEffort) {
```